src/CommunityToolkit.Maui.Core/IPopupService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.ComponentModel;
+
+namespace CommunityToolkit.Maui.Core;
+
+/// <summary>
+/// Provides a mechanism for displaying <see cref="CommunityToolkit.Maui.Core.IPopup"/>s based on the underlying view model.
+/// </summary>
+public interface IPopupService
+{
+	/// <summary>
+	/// Resolves and displays a <see cref="CommunityToolkit.Maui.Core.IPopup"/> and <typeparamref name="TViewModel"/> pair that was registered with <c>AddTransientPopup</c>.
+	/// </summary>
+	/// <typeparam name="TViewModel">The type of the view model registered with the <see cref="CommunityToolkit.Maui.Core.IPopup"/>.</typeparam>
+	void ShowPopup<TViewModel>() where TViewModel : INotifyPropertyChanged;
```

Comment:
Hi, can we use Type viewModelType instead of TViewModel in interface and make an extension with TViewModel?

Reply:
It is possible but what is your use case for wanting to supply a Type instead?

Reply:
In the codebase that I work with, unfortunately, we have a  viewModel type passing from other parts of the code, so I know the type of viewModel needed only on runtime, it  I believe I'm not the only one with such issue. Also as far as I know, it's a good practice to pass a type as a parameter and then have a generic extension

Reply:
I'd be OK with expanding the popup service to support this but perhaps we can open up a proposal to modify it once this PR gets in?